### PR TITLE
added chapter about link-provider

### PR DESCRIPTION
**Pull request by wachterjohannes**
| Q | A
| --- | ---
| Fixed tickets | none
| License | MIT

#### What's in this PR?

This PR add the chapter about LinkProvider which will be introduced by https://github.com/sulu/sulu/pull/2967

### CODE REVIEW

File: cookbook/link-provider.rst [hunk 1]
```diff
@@ -0,0 +1,195 @@
+Provider for CKeditor Internal-Sulu-Link
+========================================
+
+`LinkProvider` are used to load data for the "Internal-Sulu-Link" Plugin for
+the ckeditor.It returns an array of `LinkItem` instances, identified by ids
```
Comment (danrot) [inline cookbook/link-provider.rst]: Space after `ckeditor.`

File: cookbook/link-provider.rst [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+which will be passed to the `TeaserProviderInterface::preload` function.
+This feature can be used by the CKeditor Plugin "Sulu-Internal-Link" or
```
Comment (danrot) [inline cookbook/link-provider.rst]: Find a consisting naming for `ckeditor`.

File: cookbook/link-provider.rst [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+by adding Markup to your twig-templates (see following example and the
```
Comment (danrot) [inline cookbook/link-provider.rst]: closing `)` is missing

File: cookbook/link-provider.rst [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+        }
+    }
+
+Now you can create a service for this class and add the tag
+`<tag name="sulu.link.provider" alias="{your link-type}"/>`.
+
+How to create a custom link-select component?
+---------------------------------------------
+
+A link-select component is a simple aura-component which provides the ability
```
Comment (danrot) [inline cookbook/link-provider.rst]: `aura-component, which`

**Comment by wachterjohannes** [comment]
@danrot fixed

**Pull request merged**